File: frontend/src/shared/theme/builtinChatThemes.ts
import neonNightCityThemeJson from "../../../../assets/chat_ui_themes/neon-night-city/theme.json";
import ragingLoopSimpleThemeJson from "../../../../assets/chat_ui_themes/raging-loop-simple/theme.json";
import sakuraDreamThemeJson from "../../../../assets/chat_ui_themes/sakura-dream/theme.json";
import spiritronCommandThemeJson from "../../../../assets/chat_ui_themes/spiritron-command/theme.json";
import windborneAdventureThemeJson from "../../../../assets/chat_ui_themes/windborne-adventure/theme.json";

import { CHAT_THEME_SCHEMA, type ChatThemeManifest } from "./chatTheme";

function isRecord(value: unknown): value is Record<string, unknown> {
  return typeof value === "object" && value !== null && !Array.isArray(value);
}

function bundledThemeManifest(value: unknown): ChatThemeManifest {
  if (!isRecord(value) || value.schema !== CHAT_THEME_SCHEMA || typeof value.id !== "string") {
    throw new Error("Invalid bundled chat theme manifest");
  }
  if (!isRecord(value.name) || !isRecord(value.tokens)) {
    throw new Error(`Invalid bundled chat theme manifest: ${value.id}`);
  }

  const { $schema: _schema, ...manifest } = value;
  return manifest as unknown as ChatThemeManifest;
}

const bundledThemeManifests = [
  bundledThemeManifest(windborneAdventureThemeJson),
  bundledThemeManifest(neonNightCityThemeJson),
  bundledThemeManifest(ragingLoopSimpleThemeJson),
  bundledThemeManifest(sakuraDreamThemeJson),
  bundledThemeManifest(spiritronCommandThemeJson),
];

export const builtinChatThemeManifests: Record<string, ChatThemeManifest> = Object.fromEntries(
  bundledThemeManifests.map((manifest) => [manifest.id, manifest]),
);

File: frontend_bridge_core/builtin_chat_themes.py
"""Registry for the built-in chat UI themes.

Theme manifest content lives exclusively in ``assets/chat_ui_themes``.  This
module only keeps the stable IDs that the bridge needs for default selection
and built-in deletion protection.
"""

from __future__ import annotations


DEFAULT_BUILTIN_CHAT_THEME_ID = "windborne-adventure"
NEON_NIGHT_CITY_THEME_ID = "neon-night-city"
SAKURA_DREAM_THEME_ID = "sakura-dream"
SPIRITRON_COMMAND_THEME_ID = "spiritron-command"
RAGING_LOOP_SIMPLE_THEME_ID = "raging-loop-simple"

# These themes were already built in before per-directory ownership markers
# existed, so an unmarked directory with either ID is still managed by the
# application.  ``sakura-dream`` is intentionally absent: older releases
# allowed users to install that ID, and those directories must stay user-owned.
LEGACY_UNMARKED_BUILTIN_THEME_IDS = frozenset(
    {
        DEFAULT_BUILTIN_CHAT_THEME_ID,
        NEON_NIGHT_CITY_THEME_ID,
    }
)

BUILTIN_THEME_IDS = frozenset(
    {
        DEFAULT_BUILTIN_CHAT_THEME_ID,
        NEON_NIGHT_CITY_THEME_ID,
        SAKURA_DREAM_THEME_ID,
        SPIRITRON_COMMAND_THEME_ID,
        RAGING_LOOP_SIMPLE_THEME_ID,
    }
)
